tool/llpcAutoLayout.cpp

```diff
@@ -438,7 +444,8 @@ void DoAutoLayoutDesc(
     }
 
     // Collect ResourceMappingNode entries in sets.
-    std::map<uint32_t, std::vector<ResourceMappingNode>> resNodeSets;
+    std::map<uint32_t, ResNodeSet> resNodeSets;
```

Comment:
This code is only used in standalone amdllpc when compiling something other than a pipeline. In that configuration, in all existing tests, it only compiles a single shader. Therefore I think your suggestions are unnecessary. Can I leave it as it is?

Reply:
1. please move them to BuildPipeline or declare them as global variable.
We need use same resource node cross shader stage. especially when we compile merged shader (vs + gs) on gfx9. 
we will hit an assert in driver if the resource node is different for same resource in different shader stage 
2. please include stream out node and push constant node in the map too.

Reply:
OK, let's fix it in a separate commit.